src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -974,16 +974,24 @@ public DecisionDag(DagState rootNode)
             /// <summary>
             /// A successor function used to topologically sort the DagState set.
             /// </summary>
-            private static IEnumerable<DagState> Successor(DagState state)
+            private static ImmutableArray<DagState> Successor(DagState state)
             {
-                if (state.TrueBranch != null)
+
+                if (state.TrueBranch != null && state.FalseBranch != null)
                 {
-                    yield return state.TrueBranch;
+                    return ImmutableArray.Create(state.TrueBranch, state.FalseBranch);
                 }
-
-                if (state.FalseBranch != null)
+                else if (state.TrueBranch != null)
+                {
+                    return ImmutableArray.Create(state.TrueBranch);
+                }
+                else if (state.FalseBranch != null)
+                {
+                    return ImmutableArray.Create(state.FalseBranch);
```

Comment:
Should we cache the true and false arrays?

Reply:
Ditto.

---
In reply to: [180259808](https://github.com/dotnet/roslyn/pull/26052#discussion_r180259808) [](ancestors = 180259808)